server/executor/assertion_runner.go

```diff
@@ -103,19 +105,26 @@ func (e *defaultAssertionRunner) startWorker() {
 
 func (e *defaultAssertionRunner) runAssertionsAndUpdateResult(ctx context.Context, request AssertionRequest) (model.Run, error) {
 	log.Printf("[AssertionRunner] Test %s Run %d: Starting\n", request.Test.ID, request.Run.ID)
+
+	e.eventEmitter.Emit(ctx, model.NewTestRunEvent(model.StageTest, model.TestEventType_TestSpecsRunStart, request.Test.ID, request.Run.ID))
```

Comment:
There are some pieces of information missing here. An event should also contain a title and a description. Maybe, this is not the best way of emitting an event.

Instead of just declaring the types as constants in a file, I think it would be nice to have a package to centralize the creation of those events, so instead of:

```go
e.eventEmitter.Emit(ctx, model.NewTestRunEvent(model.StageTest, model.TestEventType_TestSpecsRunStart, request.Test.ID, request.Run.ID))
```

We could have:

```go
e.eventEmitter.Emit(ctx, events.TriggerCreated(request.Test.ID, request.Run.ID)
```

Each constructor would have a different set of parameters (testID and RunID are required in all of them) to match the description and title we want to provide.

Reply:
Make sense! Changing it!